scripts/prepdocslib/integratedvectorizerstrategy.py

```diff
@@ -0,0 +1,189 @@
+from typing import Optional
+
+# Workaround to use the preview SDK
+from azure.search.documents.indexes._generated.models import (
+    IndexProjectionMode,
+    SearchIndexerIndexProjections,
+    SearchIndexerIndexProjectionSelector,
+    SearchIndexerIndexProjectionsParameters,
+    SearchIndexerSkillset,
+)
+from azure.search.documents.indexes.models import (
+    AzureOpenAIEmbeddingSkill,
+    AzureOpenAIParameters,
+    AzureOpenAIVectorizer,
+    FieldMapping,
+    InputFieldMappingEntry,
+    OutputFieldMappingEntry,
+    SearchIndexer,
+    SearchIndexerDataContainer,
+    SearchIndexerDataSourceConnection,
+    SplitSkill,
+)
+
+from .blobmanager import BlobManager
+from .embeddings import AzureOpenAIEmbeddingService
+from .listfilestrategy import ListFileStrategy
+from .searchmanager import SearchManager
+from .strategy import DocumentAction, SearchInfo, Strategy
+
+
+class IntegratedVectorizerStrategy(Strategy):
```

Comment:
Is there logic that ensures this is only run once? Or is it idempotent? I'm having a hard time following the high level flow.

Reply:
Its idempotent.

Reply:
You set it up once and can schedule whether to run once or hour or day basis. Users have to update the blob with document and if indexer is scheduled multiple times, it will pick it up.